fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1069,6 +1069,53 @@ def assert_state(self, state: TrainingState) -> None:
             self.training_state == state
         ), f"expected to be in state {state} but current state is {self.training_state}"
 
+    @staticmethod
+    @contextlib.contextmanager
+    def enable_auto_wrap(**kwargs) -> Iterator[None]:
```

Comment:
The name `auto_wrap` made me think it is related to auto-determine which part to wrap. I suggest we rename this feature. How about calling it `config_context`?

i.e.:

with FSDP.config_context():
   FSDP.wrap_if_in_config_context(...)

Reply:
wrap_if_in_config_context might be too long, we can use wrap_if_in_CC.

Reply:
See main comments re: naming. It's really important that whatever name we choose is short and easy to type